addons/portal_rating/models/rating_rating.py

```diff
@@ -54,3 +55,15 @@ def _synchronize_publisher_values(self, values):
             if not values.get('publisher_id'):
                 values['publisher_id'] = self.env.user.partner_id.id
         return values
+
+    def _to_store_defaults(self):
+        return super()._to_store_defaults() + (
+            [
+                Store.One("message_id", []),
+                Store.One("publisher_id", ["name", "is_company", "user", "write_date"]),
+                Store.Attr("publisher_comment"),
+                Store.Attr("publisher_datetime"),
+            ]
+            if self.publisher_id
```

Comment:
_to_store_defaults is called on recordset, reading the value of a field is not ok. But as posted elsewhere, this should probably no longer be a default?

If you want to keep it as a default, then the `predicate` of Attr can be used to dynamically determine for which record to add the fields.